### PR TITLE
Improved root checking function

### PR DESCRIPTION
**Pull request by milanperera**
1. Added log messages (debug) to indicate the reason in case of failing at root checking
2. Added element to Response enum (incompatible_root)
3. Added strings to strings.xml with regard to response enum
4. Changed evaluateCompatibility() method in both DeviceStateICS and DeviceStateJB in order to return valid Response value.
5. Added a message to display in ServerDetails.java in case of failing at compatibility check.

### CODE REVIEW

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java [hunk 1]
```diff
@@ -26,6 +30,7 @@
 	private static final String[] SU_CHECK_COMMAND = new String[] { "/system/xbin/which", "su" };
 	private static final String SU_TAG = "test-keys";
 	private static final String SU_APK = "/system/app/Superuser.apk";
+    	private String TAG = Root.class.getSimpleName();
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Shouldn't this include "static final" to be a constant?

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java [hunk 2]
```diff
@@ -51,9 +55,11 @@ public boolean isDeviceRooted() {
 	 * @return - Device root status by build tags.
 	 */
 	public boolean checkRootByBuildTags() {
-		String buildTags = android.os.Build.TAGS;
-
+        	String buildTags = android.os.Build.TAGS;
 		if (buildTags != null && buildTags.contains(SU_TAG)) {
+            		if (Constants.DEBUG_MODE_ENABLED){
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: space needed after end of (Constants.DEBUG_MODE_ENABLED)

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java [hunk 3]
```diff
@@ -67,9 +73,11 @@ public boolean checkRootByBuildTags() {
 	public boolean checkRootBySuperUserApk() {
 		File suApk = new File(SU_APK);
 		if (suApk != null && suApk.exists()) {
+            		if (Constants.DEBUG_MODE_ENABLED){
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: space needed after end of (Constants.DEBUG_MODE_ENABLED)

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java [hunk 4]
```diff
@@ -80,9 +88,12 @@ public boolean checkRootBySuperUserApk() {
 	 */
 	public boolean checkRootBySuAccess() {
 		if (new ShellExecutor().executeCommand(SU_CHECK_COMMAND) != null) {
+            		if (Constants.DEBUG_MODE_ENABLED){
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: space needed after end of (Constants.DEBUG_MODE_ENABLED)

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java (hunk 1 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Fixed in PR [https://github.com/wso2/product-mdm/pull/70]

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java (hunk 2 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Fixed in PR [#70]

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java (hunk 3 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Fixed in PR [#70]

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java (hunk 4 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Fixed in PR [#70]

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java [hunk 5]
```diff
@@ -67,9 +73,11 @@ public boolean checkRootByBuildTags() {
 	public boolean checkRootBySuperUserApk() {
 		File suApk = new File(SU_APK);
 		if (suApk != null && suApk.exists()) {
```
Comment (prabathabey) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: (suApk != null) seems to be redundant.

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java [hunk 6]
```diff
@@ -581,14 +581,14 @@ public void createWebClip(String code, String data) throws AndroidAgentException
...
 			appUrl = (String) webClipData.get(resources.getString(R.string.intent_extra_identity));
 			title = (String) webClipData.get(resources.getString(R.string.intent_extra_title));
 		} catch (JSONException e) {
 			throw new AndroidAgentException("Invalid JSON format.", e);
 		}
 
-		resultBuilder.build(data);
+		resultBuilder.build(code);
 
 		if(appUrl != null && title != null){
```
Comment (prabathabey) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Formatting issues. Might not be something you've added. However, maybe a good idea to fix it anyway.

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java (hunk 5 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Fixed in PR [#70]

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java (hunk 6 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java]: Its already been fixed in [https://github.com/wso2/product-mdm/commit/b0c66506d24fb5288300bec4224f1eaa58c7b426]

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java [hunk 7]
```diff
@@ -17,6 +17,10 @@
  */
 package org.wso2.mdm.agent.api;
 
+import android.util.Log;
+
+import org.wso2.mdm.agent.utils.Constants;
+
 import java.io.File;
```
Comment (imesh) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Might be better to remove blanks lines and format the code.

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java [hunk 8]
```diff
@@ -42,7 +47,6 @@ public boolean isDeviceRooted() {
 		if (checkRootByBuildTags()) {
 			return true;
 		}
-
 		return false;
```
Comment (imesh) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: May be we can use "return checkRootByBuildTags();" directly without writing an if condition here.

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java (hunk 7 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Removed in PR [#71]

File: modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java (hunk 8 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/src/org/wso2/mdm/agent/api/Root.java]: Fixed in PR [#71]

**Pull request merged**